src/google/protobuf/string_block.h

```diff
@@ -145,7 +145,8 @@ inline StringBlock* StringBlock::Emplace(void* p, size_t n, StringBlock* next) {
   ABSL_DCHECK_EQ(n, NextSize(next));
   uint32_t doubled = static_cast<uint32_t>(n) * 2;
```

Comment:
rather than having the `static_cast` twice in this function, can you pull a local?